src/Microsoft.DotNet.Interactive/Kernel.cs

```diff
@@ -275,6 +275,24 @@ public void RegisterCommandType<TCommand>()
                             case RequestCompletions _:
                             case RequestSignatureHelp _:
                                 // FIX: (SendAsync) 
+
+                                if (_languageServiceCommandInFlight > 0 && command is RequestDiagnostics)
+                                {
+                                    context.Complete(command);
+                                }
+
+                               
+
+                                _inFlightDiagnosticsCommandInvocationContext?.Complete(_inFlightDiagnosticsCommandInvocationContext.Command);
+                                if (command is RequestDiagnostics)
```

Comment:
I would refactor so that the command types occupy different cases in the switch, and remove the need for this `if` block and the one below.

Reply:
done